### DISCUSSION TITLE
TensorFlow on Alpine

### DISCUSSION THREAD

**Discussion by EVOTk** [nextcloud/recognize#490] [Q&A] [answered]
Hello,
I am running Nextcloud using linuxserver/docker-nextcloud image based on Alpine, and TensorFlow is not available.

Do you think a solution will be possible later? Or I'd better migrate to a debian image?

Currently the facial recognition, is very very long.

![image](https://user-images.githubusercontent.com/45015615/201532222-a022c885-570e-427c-b209-2d3767dcb054.png)

Thx

**Comment by tomecp** [comment]
Could be a solution to run it as dockerfile in GPU mode: https://github.com/bugsyb/recognize_docker by @bugsyb?

**Comment by marcelklehr** [comment]
You may be able to get libtensorflow to run on alpine using the gcompat package.

**Reply by drlauridsen** [reply]
I am not very skilled with docker containers.
I installed gcompat package, any pointers what to do next?
Still says "Could not load libtensorflow in Node.js."

Running Nextcloud Linuxserver image,  which is alpine based. Recognize runs in WASM mode and is really slow.

**Reply by marcelklehr** [reply]
> Still says "Could not load libtensorflow in Node.js."

Then it probably doesn't work

**Reply by drlauridsen** [reply]
Okay thanks
I have chosen another docker image of Nextcloud instead,  havent tried facerecognition yet, but now recognize says that tensorlib is loaded, so hopefully much faster..
Just need to move my config and database..

**Comment by EthicalObligation** [comment]
I think a separate container may be necessary. This seems like the most viable solution.

**Reply by pktiuk** [reply]
This is discussed in https://github.com/nextcloud/recognize/issues/73

**Comment by bugsyb** [comment]
@EVOTk , if you don't mind switching to debian-Nextcloud based container, then as mentioned tomecp (https://github.com/nextcloud/recognize/discussions/490#discussioncomment-4700635), you might want to use dockerfiles I've shared. (might be a bit more heavy than Alpine based, though given heaviness of Nextcloud - not sure if there's much difference. 
In case if you could confirm how more lighter is the Alpine based one, I might look into customizing that container too.

**Comment by marcelklehr** [comment, marked as answer]
So, long-term the ideal path forward is to create a separate container for recognize to run models in, with the external apps ecosystem this is coming closer and closer. Short-term you can switch to a different container, ideally debian-based, and tensorflow should run there. You can also switch to a GPU-enabled container if you want, see @bugsyb 's comments.
